video_info.go

```diff
@@ -39,32 +39,32 @@ type VideoInfo struct {
 }
 
 // GetVideoInfo fetches info from a url string, url object, or a url string
-func GetVideoInfo(value interface{}) (*VideoInfo, error) {
+func (c *Client) GetVideoInfo(value interface{}) (*VideoInfo, error) {
 	switch t := value.(type) {
 	case *url.URL:
-		return GetVideoInfoFromURL(t)
+		return c.GetVideoInfoFromURL(t)
 	case string:
 		if strings.HasPrefix(t, "https://") {
 			uri, err := url.ParseRequestURI(t)
 			if err != nil {
 				return nil, err
 			}
-			return GetVideoInfo(uri)
+			return c.GetVideoInfo(uri)
 		}
 
-		return GetVideoInfoFromID(t)
+		return c.GetVideoInfoFromID(t)
 	default:
 		return nil, fmt.Errorf("Identifier type must be a string, *url.URL, or []byte")
 	}
 }
 
 // GetVideoInfoFromURL fetches video info from a youtube url
-func GetVideoInfoFromURL(u *url.URL) (*VideoInfo, error) {
+func (c *Client) GetVideoInfoFromURL(u *url.URL) (*VideoInfo, error) {
```

Comment:
Just know that you'll want to make global functions to point at the default client, lest you break every user. But, this is a draft, so I'm sure you're getting there.

Reply:
I just merged `GetVideoInfoFromURL` into `GetVideoInfo` and I added:
```golang
func GetVideoInfo(value interface{}) (*VideoInfo, error) {
	return DefaultClient.GetVideoInfo(value)
}
```